### PR TITLE
fix: remove trailing slash from blocklist client

### PR DESCRIPTION
**Pull request by matteojug**
## Description

Avoid double slashes in api calls.

## Testing Information

Tested locally with
```
[blocklist_client]
endpoint = "http://blocklist-client:3030"
```
and adding the API key.

## Checklist:

- [ ] I have performed a self-review of my code
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: signer/src/blocklist_client.rs [hunk 1]
```diff
@@ -44,11 +44,19 @@ impl BlocklistClient {
     pub fn new(ctx: &impl Context) -> Option<Self> {
         let config = ctx.config().blocklist_client.as_ref()?;
 
-        let config = Configuration {
+        let mut config = Configuration {
```
Comment (cylewitruk) [inline signer/src/blocklist_client.rs:47]: I know this isn't your doing, but it feels pretty extreme that we're talking a whole `Context` to build this when all we really need is the url -- makes the tests below a lot more complicado than they need to be 😬

**Review by cylewitruk — APPROVED** [review]
Looks good, with the exception of that lingering `Context` thing that I mentioned (but has nothing to do with these changes)

File: signer/src/blocklist_client.rs [hunk 2]
```diff
@@ -44,11 +44,19 @@ impl BlocklistClient {
...
             base_path: config.endpoint.to_string(),
             ..Default::default()
         };
 
+        // Url::parse defaults `path` to `/` even if the parsed url was without the trailing `/`
```
Comment (aldur) [inline signer/src/blocklist_client.rs:52]: I was just wondering who I had to blame for this behavior, stdlib that thinks it is smarter than us!

**Review by aldur — APPROVED** [review]

File: signer/src/blocklist_client.rs (hunk 1 above)
Comment (matteojug) [reply signer/src/blocklist_client.rs:47]: Yeah, I will fix this later

**Pull request merged**